packages/kit/src/exports/vite/build/build_service_worker.js

```diff
@@ -35,21 +35,25 @@ export async function build_service_worker(
 	fs.writeFileSync(
 		service_worker,
 		`
+			export const base = /*@__PURE__*/ location.pathname.split('/').slice(0, -1).join('/');
```

Comment:
Having a hard time understanding how/why this works. If you you visit `example.com/foo/bar` wouldn't the base be `/foo`, even if I _didn't_ set a base path?

Reply:
In a service worker, `location` is the location of the service worker, _not_ the URL you're currently visiting. Because SvelteKit guarantees that the service worker is at the root of the app (because a service worker can only handle requests at the same level or below), `location.pathname.split('/').slice(0, -1).join('/')` is guaranteed to be the base path

Reply:
Thanks for clearing it up!